### ISSUE TITLE
"editor.suggestSelection": "recentlyUsedByPrefix": is not working

### ISSUE THREAD

**Issue by svipben** [Microsoft/vscode#42795]
- VSCode Version: 1.20.0 (latest Insiders)
```
f2b21097100fc2fc0e84cc34f453c9a85a49a2bb
2018-02-02T06:03:34.306Z
```
- OS Version: macOS

`"editor.suggestSelection": "recentlyUsedByPrefix",` works the same as `recentlyUsed`.

![image](https://user-images.githubusercontent.com/10441177/35721035-ee6664ca-07f8-11e8-8441-004cac178351.png)

![image](https://user-images.githubusercontent.com/10441177/35721052-ff48e66e-07f8-11e8-8b2d-5658404dafaf.png)

![image](https://user-images.githubusercontent.com/10441177/35721065-0d5ff242-07f9-11e8-8960-f093827cc87d.png)

Description says:
```
recentlyUsedByPrefix: Select suggestions based on previous prefixes that have completed those suggestions, e.g. `co -> console` and `con -> const`.
```

But I tested and it seems it's the same as:
```
recentlyUsed: Select recent suggestions unless further typing selects one, e.g. `console.| -> console.log` because `log` has been completed recently.
```

TL;DR: `recentlyUsedByPrefix` **=** `recentlyUsed`, `recentlyUsed` **=** `recentlyUsedByPrefix`

cc @jrieken

**Comment by svipben** [comment]
Sorry @bpasero I assigned you, somehow I forgot it's @jrieken feature. 😄

**Comment by svipben** [comment]
@jrieken  I believe the problem is here: https://github.com/Microsoft/vscode/commit/80a11d95c711424ca88c898f2e52a80604fcdb10#diff-7dc5ab2f56440232a3ea552c9399b1a1R834

You forgot to change also in code, as I see you only changed names here: `src/vs/editor/common/config/commonEditorConfig.ts`.

**Comment by svipben** [comment]
@bpasero I see @jrieken was active before 5 days should I open PR with fixes? Because otherwise this will be inside stable release (1.20.0).

**Comment by bpasero** [comment]
@svipben I think @jrieken is on vacation so we will probably ship with this, but feel free to open a PR.

**Comment by svipben** [comment]
@bpasero Thanks for info then I will open a PR.
